### DISCUSSION TITLE
Couple Ideas/Observations

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [dvmarinoff/Auuki#128] [Ideas]
I've been using the web app for the last few weeks...It works great.  I actually put it on a local server to test it out, which also works great.  I'm not sure the web app is going to be developed anymore but if it were the ability to send a completed ride to Strava would be great.

Also the power graph - when it get's to the edge of the screen - the right edge, if it were able to jump back to the middle of the screen or 1/4 back so you can see the graph building better than right at the edge.

Otherwise this is great - I might keep using it for its simplicity compared to all the other apps out there.

**Comment by dvmarinoff** [comment]
Hey, Thanks!

I know it seems quiet right now, but there are actually multiple things under development.
I’ve been working on a backend API since this Spring. When it gets ready it will allow integrations with other platforms.
I am planning to add support for intervals.icu first, because I found my self using it much more than Strava, so Strava will be second, and then maybe Garmin Connect.

The thing is … I got distracted with trying to learn Rust and build the API in Rust, which led to me trying out like 4 or 5 different Rust Web frameworks, and experimenting with stuff. Then I decided that the eco system of Rust is just not suitable/ready for web development. The type system is very cool but when you have a million complex dependencies stuff gets nasty.
So I just wrote the thing in NodeJS, but decided to finally learn GraphQL, so … I am doing that now.

Parallel to this I am building an iOS app for Flux, which means I am also learning Swift and SwiftUI, and this takes time.

There are many things I want to add to the current web interface:
- a simple 3D navigation style graphics for Course simulation,
- Interface for configuring the various simulation parameters like, wind speed and direction, crr, CdA, etc.,
- cornering + virtual brakes for speed simulation,
- visual editor for .fit files that will allow to fix GPS track and elevation errors,
- Integrate the .zwo editor into the workout list, for editing in place the uploaded workouts,
- finishing the support for ANT+,

But at the moment any big feature changes are on freeze until I finish with the backend API and the iOS app.

About the power graph I’ve tried before to start it from the start and from the middle, but in the end I just didn’t like it.
Since you already run it locally, if you wish to fork the project on GitHub I can make a PR to your fork that jumps the graph back to the middle or 1/4.

**Reply by Lunchtime0614** [reply]
No worries - I wish I had the skills to modify this on my own, but it's just not where my skillset lies.  To be clear - the web app is fantastic and I plan on continuing to use it as is.  Love seeing what you have planned, and totally understand the time it takes to build these things!

**Reply by Lunchtime0614** [reply]
I also use Intervals.icu primarily so that's good news.  However I also use RunGap app on iOS to aggregate and share from one platform to the next - mainly to get all workouts into intervals.  So I wouldn't care about the strava integration if it went direct to intervals...that's my end goal anyway!!
